metric2/sync/int64/syncint64metric.go

```diff
@@ -0,0 +1,67 @@
+package syncint64metric
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	metric "go.opentelemetry.io/otel/metric2"
+)
+
+// TODO instrument options
+
+type Meter struct {
+}
+
+type Counter struct {
+}
+
+type UpDownCounter struct {
+}
+
+type Histogram struct {
+}
+
+type Instrument interface {
+	metric.Instrument
+
+	Measure(x int64) metric.Measurement
+}
+
+var (
+	_ Instrument = Counter{}
+	_ Instrument = UpDownCounter{}
+	_ Instrument = Histogram{}
+)
+
+func (m Meter) Counter(name string) (Counter, error) {
+	return Counter{}, nil
+}
+
+func (m Meter) UpDownCounter(name string) (UpDownCounter, error) {
+	return UpDownCounter{}, nil
+}
+
+func (m Meter) Histogram(name string) (Histogram, error) {
+	return Histogram{}, nil
+}
+
+func (c Counter) Add(ctx context.Context, x int64, attrs ...attribute.KeyValue) {
+}
+
+func (u UpDownCounter) Add(ctx context.Context, x int64, attrs ...attribute.KeyValue) {
+}
+
+func (h Histogram) Record(ctx context.Context, x int64, attrs ...attribute.KeyValue) {
+}
+
+func (c Counter) Measure(x int64) metric.Measurement {
+	return metric.Measurement{}
+}
+
+func (u UpDownCounter) Measure(x int64) metric.Measurement {
```

Comment:
Yes. Measure() creates a measurement for batch recording in both synchronous and asynchronous instruments, and it can be added later. My goal in sharing this prototype was to show how it would evolve to support batch measurements since that is an important way to optimize metrics costs. The other way that we know of to optimize at the API level involves binding instruments, which I left out of this proposal (and I consider less important than batch support).